### DISCUSSION TITLE
Can you create array in Slint programatically - list comprehension?

### DISCUSSION THREAD

**Discussion by JakubKoralewski** [slint-ui/slint#5911] [Q&A] [answered]
I have a property that passes in a Struct in an array, and a ComboBox with `model:` that requires values from each individual struct's property `name`.
```slint
struct Struct {
    name: string
}
component ProblematicRepetition {
    in property <[Struct]> structs;
	ComboBox {
		model: ??? // structs.map(|s| s.name) OR structs.name OR [s.name for s in structs] OR [for s in structs : s.name];
	}
}
```

Currently I have to do this to work around it:
```slint
struct Struct {
    name: string
}
component ProblematicRepetition {
    in property <[string]> struct-names;
    in property <[Struct]> structs;
	ComboBox {
		model: struct-names
	}
}
```
But I'd like to avoid the duplication ideally. Is there any way to do this right now? If not, can such feature be added?

**Comment by tronical** [comment, marked as answer]
At the moment this is not possible in pure Slint code. We're tracking this feature in https://github.com/slint-ui/slint/issues/1328 . So at the moment the best solution is to use a `MapModel` in native code if you're using Rust or C++.
